### PR TITLE
Kubernetes

### PR DESCRIPTION
**Pull request by colinbruce**
#### What
Update application for deploying into the Kubernetes cluster

#### Ticket
[CCCD: Set up dev K8s environment](https://dsdmoj.atlassian.net/browse/CBO-663)

#### Why
The long term plan for hosting is for everything to be in the K8s cluster, therefore this work is future-proofing CCCD against Template Deploy reaching end of life

#### How
* Update the .files to streamline the process
* Add scripts to assist the debugging needed for the transition
* Add the `kubectl` deploy scripts needed to deploy to the new hosting environments

---

#### TODO (wip)

- [x] Add dev, staging and api-sandbox environments
- [x] implement DI (SNS/SQS) in dev
- [x] implement DI (SNS/SQS) in staging
- [x] git-crypted secrets config
- [x] binary for retrieval and application of secrets
- [x] config and scripts for build and deploy of app
- [x] config and scripts for migrate and seed DB jobs
- [x] config and scripts for cronjobs (archiving/destroying stale claims  + delete untagged ECR images)
- [x] move all scripts/binaries under kubernetes directory

--- 
#### Related PRs
https://github.com/ministryofjustice/advocate-defence-payments-deploy/pull/187

---
#### Merge preparation
- [x] ensure s3 working in all k8s and TD environments
- [x] ensure DI (SNS/SQS) working in dev k8s environments (k8s staging requires wiring up to UAT)
- [x] ensure DI (SNS/SQS) working in demo and staging TD environments
- [ ] merge to master
--- 
#### Additional work
- [ ] implement db jobs (migrate/seed) as [optional] part of deploy
- [ ] amend SQS terraform to create fixed queue names that can be used in CCCD, CCR and CCLF
- [ ] test and document S3 bucket move/migration
- [ ] test and document RDS db move/migration

### CODE REVIEW

**Comment by jsugarman** [comment]
@colinbruce be worth you reviewing the last few commits of this branch at least. Have moved the scripts and generated binaries from them. be good if you could pull and verify whether, for example, `kubernetes_deploy/bin/deploy dev kubernetes-latest` works for you (does for me).

til thursday 👋

File: README.md [hunk 1]
```diff
@@ -294,6 +289,26 @@ The claim's provider has an attribute 'vat_registered' which governs whether or
 
 For both VAT registered and unregistered LGFS providers, a VAT amount field is provided for manual input of VAT
 
+### Kubernetes
+
+Config for kubernetes object can be found under `kubernetes_deploy/`. In addition binaries in `kubernetes_deploy/bin` have been generated from scripts in `kubernetes_deploy/scripts`.
```
Comment (colinbruce) [inline README.md:294]: I'm not clear on the benefits of `regenerating binaries` but this looks impressive! :)
Are the `*.sh.x.c` part of this too?

**Review by colinbruce — COMMENTED** [review]
Looks good to me :+1:

File: README.md (hunk 1 above)
Comment (jsugarman) [reply README.md:294]: yes, they are what is generated by `shc`. i was of two minds whether to bother and we can easily remove the binaries and rely on `kubernetes_deploy/scripts/build.sh` instead if the overhead is tedious

the flow for the binaries is just amend the script then run
`shc -f  kubernetes_deploy/script/build.sh -o kubernetes_deploy/bin/build`
to regenerate the binary

**Review by jsugarman — APPROVED** [review]
👍 on behalf of @colinbruce who reviewed above

**Pull request merged**